### PR TITLE
Create sparql-reference.md

### PR DESCRIPTION
**Pull request by shawntanzk**
made a start on this, will do this bit by bit

### CODE REVIEW

File: docs/reference/sparql-reference.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+# adding prefixes used
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix obo: <http://purl.obolibrary.org/obo/>
+
+# selecting 2 variables, prefix and numberOfClasses, where number of classes is a count of distinct cls
+SELECT ?prefix (COUNT(DISTINCT ?cls) AS ?numberOfClasses) WHERE 
+{
+  # the variable cls is a class 
+  ?cls a owl:Class .
+  # removes any cases where the variable cls is blank
```
Comment (shawntanzk) [inline docs/reference/sparql-reference.md:39]: @matentzn - dont quite understand this step, why is it needed?

Comment (matentzn) [reply docs/reference/sparql-reference.md:39]: Which step, the "owl:Class" step? This is to ensure the query is not looking for object properties...

Comment (shawntanzk) [reply docs/reference/sparql-reference.md:39]: oh no, the isblank step - just asked anita (we are in the office today) she said something to the extent of blank classes get "created" in triple stores, and that was to exclude. What I am a bit confused I guess is 1) why that is needed with the group by, and 2) why when i run this, there is still a blank row

Comment (matentzn) [reply docs/reference/sparql-reference.md:39]: Ask us on Monday about blank classes, suffices to say: some nodes in RDF, like existential restriction, do not have an ID. So we want to ensure that these nodes are excluded from the query!

**Comment by shawntanzk** [comment]
Think this is a good start, we can add more bit by bit later - would like a review that i didnt say anything stupid though lol

**Comment by shawntanzk** [comment]
SPARQL basics needs a bit of relooking at - will add it to the board

File: docs/reference/sparql-reference.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  FILTER(isIRI(?term) && (STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/MONDO_")))
+}
+```
+
+### Definition lacks xref 
+adaptable for lacking particular annotation
+
+```SPARQL
+# adding prefixes used
+prefix oio: <http://www.geneontology.org/formats/oboInOwl#>
```
Comment (matentzn) [inline docs/reference/sparql-reference.md:29]: always use `prefix oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>`, not `oio`

File: docs/reference/sparql-reference.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  FILTER NOT EXISTS {
+    ?def_anno oio:hasDbXref ?x .
+  }
+  
+  # removes triples where entity is blank
+  FILTER (!isBlank(?entity))
+  # selects entities that are native to ontology (in this case MONDO)
+  FILTER (isIRI(?entity) && STRSTARTS(str(?entity), "http://purl.obolibrary.org/obo/MONDO_"))
+  # assigning def: as the property variable
+  BIND(def: as ?property)
```
[the remark this answers is not in the document]
Comment (matentzn) [reply docs/reference/sparql-reference.md:54]: Like n the query below!

File: docs/reference/sparql-reference.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix IAO: <http://purl.obolibrary.org/obo/IAO_>
+
+# selecting only unique instances of the three variables 
+SELECT DISTINCT ?entity ?property ?value WHERE 
+{
+  # the variable property has to be IAO:0000115 
+  VALUES ?property {
+    IAO:0000115
```
Comment (matentzn) [inline docs/reference/sparql-reference.md:74]: here you use IAO:0000115, above in the query you use `def:` for the same thing. Maybe lets universally settle on `definition:`?

File: docs/reference/sparql-reference.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+prefix owl: <http://www.w3.org/2002/07/owl#>
+prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+prefix oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+prefix IAO: <http://purl.obolibrary.org/obo/IAO_>
+prefix RO: <http://purl.obolibrary.org/obo/RO_>
+prefix mondo: <http://purl.obolibrary.org/obo/mondo#>
+prefix skos: <http://www.w3.org/2004/02/skos/core#>
+prefix dce: <http://purl.org/dc/elements/1.1/>
+prefix dc: <http://purl.org/dc/terms/>
```
Comment (matentzn) [inline docs/reference/sparql-reference.md:100]: use dcterms, not dc

File: docs/reference/sparql-reference.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+```SPARQL
+
+# adding prefixes used
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
+
+# selecting only unique instances of the three variables 
+SELECT DISTINCT ?entity ?property ?value WHERE {
+ # the variable property is IAO_0100001 (item replaced by)
+ VALUES ?property { <http://purl.obolibrary.org/obo/IAO_0100001> }
```
Comment (matentzn) [inline docs/reference/sparql-reference.md:130]: Make a nice named prefix for IAO_0100001

File: docs/reference/sparql-reference.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+# selecting 2 variables, prefix and numberOfClasses, where number of classes is a count of distinct cls
+SELECT ?prefix (COUNT(DISTINCT ?cls) AS ?numberOfClasses) WHERE 
+{
+  # the variable cls is a class 
+  ?cls a owl:Class .
+  # removes any cases where the variable cls is blank
+  FILTER (!isBlank(?cls))
+  # Binds the variable prefix as the prefix of the class (eg. MONDO, CL, etc.)
+  BIND( STRBEFORE(STRAFTER(str(?cls),"http://purl.obolibrary.org/obo/"), "_") AS ?prefix)
```
Comment (matentzn) [inline docs/reference/sparql-reference.md:161]: you should add a FILTER for ?cls that STRSTARTS with the OBO purl blurp

File: docs/reference/sparql-reference.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (shawntanzk) [reply docs/reference/sparql-reference.md:54]: honestly, i just copied queries and wrote comments on them, so if this is here, its somewhere out there lol, but will change

[the remark this answers is not in the document]
Comment (matentzn) [reply docs/reference/sparql-reference.md:54]: Yeah I know! The wild is a bit more chaotic though, here we have an opportunity to really unify the way we write sparql queries!

File: docs/reference/sparql-reference.md (hunk 7 above)
Comment (shawntanzk) [reply docs/reference/sparql-reference.md:161]: not quite sure what this means?

Comment (shawntanzk) [reply docs/reference/sparql-reference.md:166]: oh i get it, actually, I quite like that there is a blank to tell me that they are not obo purls

**Review by matentzn — APPROVED** [review]

**Pull request merged**